File: Cuarto/Laboratorio/Talleres/Entrega1/CombatStrategy.java
package Robotito;

import robocode.*;

public interface CombatStrategy {
    double bulletHalfPower = 1.5;
    double bulletMaxPower = 3;

    void init(JuniorRobot robot);

    void onTick(JuniorRobot robot);

    void onScannedRobot(JuniorRobot robot);

    void onHitWall(JuniorRobot robot);

    void onHitRobot(JuniorRobot robot);

    void onHitByBullet(JuniorRobot robot);

    default boolean readyToAttack(){
        return true;
    }

}

File: Cuarto/Laboratorio/Talleres/RobocodeLabo-kotlin/src/Robotito/CombatStrategy.kt
package Robotito

import robocode.*;

abstract class CombatStrategy {

    protected lateinit var robot: JuniorRobot
    protected val bulletHalfPower = 1.5
    protected val bulletMaxPower = 3.0

    fun init(robot: JuniorRobot) {
        this.robot = robot
        onInit()
    }

    abstract fun onInit()
    abstract fun onTick()
    abstract fun onScannedRobot()
    abstract fun onHitWall()
    abstract fun onHitRobot()
    abstract fun onHitByBullet()

    open fun readyToAttack(): Boolean {
        return true
    }
}
